### DISCUSSION TITLE
Ozone development environment questions

### DISCUSSION THREAD

**Discussion by Milind220** [Ozon3Org/Ozon3#82] [General]
I noticed that the new setup guide has a command to install the package in editable mode. I'm curious, what version of Ozone does this make it install? I'm assuming it installs the latest released version - is this correct?

@lahdjirayhan

**Comment by Samxx97** [comment]
@Milind220 editable installs are always used for development purposes, they install the code that is present in the current directory where you invoke the command `pip install -e . ` and so i assume it searches the directory for any packaging info files like pyproject.toml and setup.cfg and for that reason yes they do install the latest version provided that the contributor who is running this command pulled in the latest code with the latest version from the repository, also the cool thing about it, is the fact that once u run this command the package is installed in your virtual environment and any later changes you make to the code will take effect automatically and you won't have to reinstall package after every small change you make, and they are also very useful for separating tests into its own folder since after you installed the package in your virtual environment  the tests can access (import) your core functionality from anywhere and it doesn't have to resort to absolute or relative imports.

**Reply by lahdjirayhan** [reply]
This answer has perfectly summarized it. I recommend using editable mode (instead of installing a fixed version of Ozone) because of precisely this:
> any later changes you make to the code will take effect automatically and you won't have to reinstall package after every small change you make

When developing, it is a very convenient thing to have.

In addition, if you want to have specific Ozone version after you installed it in editable mode, you can always just checkout the version tag. Or if you want to have latest Ozone release, just checkout main.

**Reply by Milind220** [reply]
Ahh okay that makes a lot of sense. I really like that part about the tests, that was kinda what I was getting at. Thanks!
